### ISSUE TITLE
unable to clone repository

### ISSUE THREAD

**Issue by jape42** [PABannier/bark.cpp#130]
git clone --recursive https://github.com/PABannier/bark.cpp.git
Cloning into 'bark.cpp'...
remote: Enumerating objects: 700, done.
remote: Counting objects: 100% (360/360), done.
remote: Compressing objects: 100% (145/145), done.
remote: Total 700 (delta 292), reused 224 (delta 214), pack-reused 340
Receiving objects: 100% (700/700), 47.85 MiB | 10.92 MiB/s, done.
Resolving deltas: 100% (390/390), done.
Submodule 'encodec.cpp' (https://github.com/PABannier/encodec.cpp) registered for path 'encodec.cpp'
Cloning into '/mnt/ubuntu/home/jape/ai/bark.cpp/encodec.cpp'...
remote: Enumerating objects: 275, done.        
remote: Counting objects: 100% (122/122), done.        
remote: Compressing objects: 100% (64/64), done.        
remote: Total 275 (delta 84), reused 68 (delta 52), pack-reused 153        
Receiving objects: 100% (275/275), 3.93 MiB | 9.86 MiB/s, done.
Resolving deltas: 100% (155/155), done.
fatal: remote error: upload-pack: not our ref e50cd96d28c89f6c1343c291042b14bab6f3b83b
fatal: Fetched in submodule path 'encodec.cpp', but it did not contain e50cd96d28c89f6c1343c291042b14bab6f3b83b. Direct fetching of that commit failed.

**Comment by montecarlo26** [comment]
I am experiencing the same issue with one additional caveat...

No VPN:
```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Install the latest PowerShell for new features and improvements! https://aka.ms/PSWindows

PS C:\Users\Quill\Documents\ai\text to speech\bark 5> git clone --recursive https://github.com/PABannier/bark.cpp.git
Cloning into 'bark.cpp'...
remote: Enumerating objects: 700, done.
remote: Counting objects: 100% (351/351), done.
remote: Compressing objects: 100% (147/147), done.
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
error: 807 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
PS C:\Users\Quill\Documents\ai\text to speech\bark 5>
```
VPN:
```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Install the latest PowerShell for new features and improvements! https://aka.ms/PSWindows

PS C:\Users\Quill\Documents\ai\text to speech\bark 5> git clone --recursive https://github.com/PABannier/bark.cpp.git
Cloning into 'bark.cpp'...
remote: Enumerating objects: 700, done.
remote: Counting objects: 100% (351/351), done.
remote: Compressing objects: 100% (147/147), done.
remote: Total 700 (delta 285), reused 213 (delta 203), pack-reused 349
Receiving objects: 100% (700/700), 47.85 MiB | 28.75 MiB/s, done.
Resolving deltas: 100% (391/391), done.
Submodule 'encodec.cpp' (https://github.com/PABannier/encodec.cpp) registered for path 'encodec.cpp'
Cloning into 'C:/Users/Quill/Documents/ai/text to speech/bark 5/bark.cpp/encodec.cpp'...
remote: Enumerating objects: 275, done.
remote: Counting objects: 100% (122/122), done.
remote: Compressing objects: 100% (64/64), done.
remote: Total 275 (delta 84), reused 68 (delta 52), pack-reused 153
Receiving objects: 100% (275/275), 3.93 MiB | 8.32 MiB/s, done.
Resolving deltas: 100% (155/155), done.
fatal: remote error: upload-pack: not our ref e50cd96d28c89f6c1343c291042b14bab6f3b83b
fatal: Fetched in submodule path 'encodec.cpp', but it did not contain e50cd96d28c89f6c1343c291042b14bab6f3b83b. Direct fetching of that commit failed.
PS C:\Users\Quill\Documents\ai\text to speech\bark 5>
```
Git or my ISP (Cox) is actually throttling clones of this repository, causing an error (Kbps vs  Mbps). Using a VPN fixes it. I don't know what the motive is behind this, hopefully it's a mistake. I will assume this issue is specific to my IP address but I'm reporting it anyways in case it isn't.

In any case I get further when using a VPN until I encounter the same ref issue you reported:
```
fatal: remote error: upload-pack: not our ref e50cd96d28c89f6c1343c291042b14bab6f3b83b
```

**Comment by PeterCho** [comment]
I have run into the same problem as jape42 reported a few days ago.

**Comment by jmtatsch** [comment]
+1

**Comment by Runtrons** [comment]
+1

**Comment by mattkanwisher** [comment]
```(base) ➜  encodec.cpp git:(main) ✗ git checkout e50cd96d28c89f6c1343c291042b14bab6f3b83b.
error: pathspec 'e50cd96d28c89f6c1343c291042b14bab6f3b83b.' did not match any file(s) known to git```

yeah this commit no longer exists in the Encodec repo

**Comment by mattkanwisher** [comment]
easy fix
```bash
cd encodec.cpp
git checkout 80799dd6773b3f5d017bbe0597d1ee141302af8d 
git submodule update --init --recursive
```

(or whatever the latest commit is when you look)

**Comment by PlanetMacro** [comment]
same

**Comment by PABannier** [comment]
Hi All,

This should be fixed with #139 . Could you give it another try?
